src/main/java/com/rocketden/main/util/Utility.java

```diff
@@ -2,23 +2,58 @@
 
 import java.util.Random;
 
+import com.rocketden.main.dao.RoomRepository;
+import com.rocketden.main.dao.UserRepository;
+
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class Utility {
 
     private static final Random random = new Random();
+    public static final String ROOM_ID_KEY = "ROOM_ID";
+    public static final String USER_ID_KEY = "USER_ID";
+
+    private final RoomRepository roomRepository;
+    private final UserRepository userRepository;
+
+    @Autowired
+    public Utility(RoomRepository roomRepository, UserRepository userRepository) {
+        this.roomRepository = roomRepository;
+        this.userRepository = userRepository;
+    }
 
     // Generate numeric String with a specific length.
-    public String generateId(int length) {
+    public String generateUniqueId(int length, String idType) {
         String numbers = "1234567890";
         char[] values = new char[length];
-
-        for (int i = 0; i < values.length; i++) {
-            int index = random.nextInt(numbers.length());
-            values[i] = numbers.charAt(index);
+        String id = null;
+        
+        /**
+         * Continue generating new id values until a valid one is provided.
+         * Note: This loop could technically run forever, but there are 9^6
+         * (531,441) possible combinations, so this works for now (10/20/2020).
+         */
+        while (id == null || idExists(id, idType)) {
```

Comment:
Nit: I think checking if an ID exists could be simplified to a ternary statement inside the loop that sets `id` to `values` or `null`. The code still works the same way so it's not required but you save a line of code I guess?

Reply:
If I understand correctly, this would just move the ID validation to within the loop (and have it continually `null` until it reaches a valid value) instead of outside of the loop; since I'd still need the check `id == null` in the while loop, I don't think this would save any lines? Since I think it's more intuitive to include all of the checking within the while condition, I'll leave it as-is for now; however, please let me know if you still think I should change this.